src/main/java/duke/tasklist/TaskList.java

```diff
@@ -0,0 +1,193 @@
+package duke.tasklist;
+
+import duke.exception.DukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Locale;
+
+/**
+ * Represents the list of tasks.
+ */
+public class TaskList {
+
+    private static ArrayList<Task> tasks;
+    private static final DateTimeFormatter FORMAT_TIME = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+    private static final DateTimeFormatter FORMAT_NO_TIME = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+
+    /**
+     * Constructor for TaskList.
+     *
+     * @param tasks ArrayList of Tasks.
+     */
+    public TaskList(ArrayList<Task> tasks) {
+        this.tasks = tasks;
+    }
+
+    /**
+     * String of all the tasks.
+     *
+     * @return Returns a list of all tasks.
+     */
+    public String getAllTasksString() {
+        String output = "";
+        for (int i = 0; i < tasks.size(); i++) {
+            output += i + 1 + ": " + tasks.get(i) + "\n";
+            System.out.println(i + 1 + ": " + tasks.get(i));
+        }
+        return output;
+    }
+
+    /**
+     * String of a task with the index specified.
+     *
+     * @param index Index of the task to print.
+     * @return Task with the specified index.
+     */
+    public String getTaskString(int index) { // RENAME
+        System.out.println(index + 1 + ": " + tasks.get(index));
+        return index + 1 + ": " + tasks.get(index);
+    }
+
+    /**
+     * Mark a task as done.
+     *
+     * @param taskId ID of the task we are marking as done.
+     * @return Response to the task being marked as done.
+     */
+    public String markTaskAsDone(int taskId) throws DukeException {
+        if (taskId < 1 || taskId > tasks.size()) {
+            throw new DukeException("☹ OOPS!!! Please provide a task ID that exists.");
+        }
+        tasks.get(taskId - 1).setIsDone(true);
+        System.out.println("Nice! I've marked this task as done.");
+        System.out.println("   " + tasks.get(taskId - 1));
+        return "Nice! I've marked this task as done.\n" + "    " + tasks.get(taskId - 1);
+    }
+
+    /**
+     * Adds a duke.task.Todo task.
+     *
+     * @param description Description of the duke.task.Todo.
+     * @return Response to a Todo being added.
+     */
+    public String addTodo(String description) {
+        tasks.add(new Todo(description, false));
+        return getLatestTaskString();
+    }
+
+    /**
+     * Adds a duke.task.Deadline task.
+     *
+     * @param fullDescription String that contains the description and deadline of the task.
+     * @return Response to a deadline being added.
+     */
+    public String addDeadline(String fullDescription) throws DukeException {
+        int sepIndex = fullDescription.indexOf("/by");
+        if (fullDescription.charAt(sepIndex + 3) != ' ' || sepIndex == -1 || sepIndex == 0 ||
```

Comment:
nit: this seems like a fairly complex expression that could be broken down and given meaningful names to help a little with code comprehension